src/lib/challenge.ts

```diff
@@ -48,3 +52,21 @@ export async function generate (size: number = 32): Promise<string> {
     throw error
   }
 }
+
+/*
+ * Helper function to validate signatures using public key
+ * @param challenge Base64 challenge string
+ * @param signature Base64 sign string
+ * @param publicKey Base64 RSA/EC:sec256k1 key string or KeyObject for verification
+ */
+export async function verifySign (
```

Comment:
Is there any need for this to be async? I don't see any async functions being called here.

Reply:
@lewisdaly that's something I wanted to confirm with you and @eoln . The `crypto` function for` generateChallenge` is promisified to make the overall function async. Should I also make this function aync?